logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -312,21 +321,75 @@ function New-DevelopmentAppSettings([hashtable] $Settings = @{ }) {
         $developmentSettingsForEngine = (Get-DefaultDevelopmentSettingsByEngine)[$Settings.ApiSettings.Engine]
         $developmentSettingsForEngine = Add-ApplicationNameToConnectionStrings $developmentSettingsForEngine $project
 
-        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine, $Settings
+        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine
 
         $newDevelopmentSettings = Add-TestHarnessSpecificAppSettings $newDevelopmentSettings $project
 
+        $newDevelopmentSettings = Merge-Hashtables $newDevelopmentSettings, (Get-CredentialSettingsByProject)[$project], $Settings
+
         $projectPath = Get-RepositoryResolvedPath $Project
 
         $newDevelopmentSettingsPath = Join-Path $projectPath "appsettings.development.json"
         New-JsonFile $newDevelopmentSettingsPath $newDevelopmentSettings -Overwrite
         $newSettingsFiles += $newDevelopmentSettingsPath
 
         $credentialSettings = Merge-Hashtables @{ }, (Get-CredentialSettingsByProject)[$project]
-        $newUserSettingsPath = Join-Path $projectPath "appsettings.user.json"
-        New-JsonFile $newUserSettingsPath $credentialSettings
-        $newSettingsFiles += $newUserSettingsPath
+        $userSecrets += ($credentialSettings | ConvertTo-Json -Depth 10 | dotnet user-secrets set --project $projectPath --id (Get-UserSecretsIdByProject).$project)
     }
 
     return $newSettingsFiles
 }
+
+$RegExFilter = "\A(?<curr>[^:]+)(:(?<remain>(?<sub>[^:]+)(:.*)?))?\Z"
+
+function Get-FlattenHash {
+    param([string]$Key,$Value,$Obj)  
+
+    if ($Value) {
+        if ($key -match $RegExFilter) {
+            $Current = $matches.curr
+            if (-not ($Obj | gm $Current)) {
+                        if (-not $Obj.ContainsKey($Current)) { $Obj.$Current = @{ } }
+            }
+
+            if ($matches.remain) {
+                $Obj.$Current = Get-FlattenHash -Key $matches.remain -Value $Value -Obj ($Obj.$Current)
+            } else {
+                $Obj.$Current = $Value
+            }
+            return $Obj
+        }
+    }
+}
+
+function Get-UserSecrets() {
+    $developmentSettingsByProject = Get-DefaultDevelopmentSettingsByProject
+    $inputTable = @{}
+    foreach ($project in $developmentSettingsByProject.Keys){
+        $projectPath = Get-RepositoryResolvedPath $project
+        $userSecretList= dotnet user-secrets list --project $projectPath --id (Get-UserSecretsIdByProject).$project | Out-String
```

Comment:
I mentioned this a couple times in slack but since we are replacing the `Application\EdFi.Ods.WebApi.NetCore\appsettings.user.json` with these user secrets I'm not sure if it makes sense, or will have the intended outcome we want, to have user secrets for all the projects being pulled into initdev. We should probably make this only use the EdFi.Ods.WebApi.NetCore user-secrets.

Reply:
Now changed to getting EdFi.Ods.WebApi.NetCore user-secrets only here .